artio-binary-entrypoint-impl/src/main/java/uk/co/real_logic/artio/binary_entrypoint/InternalBinaryEntryPointConnection.java

```diff
@@ -693,7 +693,10 @@ else if (sessionVerID != this.sessionVerId)
 
     protected Action unbindState(final DisconnectReason reason)
     {
-        cancelOnDisconnect.checkCancelOnDisconnectDisconnect();
+        if (!DisconnectReason.LOGOUT.equals(reason))
```

Comment:
`unbindState` will disconnect the client and enqueue task to trigger `onCancelOnDisconnectTimeout`, but should not do this when the client disconnects gracefully (logout) while configured with `CANCEL_ON_DISCONNECT_ONLY`